src/basic/virt.c

```diff
@@ -227,12 +227,31 @@ static int detect_vm_xen(void) {
         return VIRTUALIZATION_XEN;
 }
 
+#define XENFEAT_dom0 11 /* xen/include/public/features.h */
 /* Returns -errno, or 0 for domU, or 1 for dom0 */
 static int detect_vm_xen_dom0(void) {
         _cleanup_free_ char *domcap = NULL;
+        static const char path_features[] = "/sys/hypervisor/properties/features";
         char *cap, *i;
         int r;
 
+        r = read_one_line_file(path_features, &domcap);
+        if (r < 0 && r != -ENOENT)
+                return r;
+	if (r == 0) {
+                unsigned long features = 0;
```

Comment:
No need to initialize this.
Also please add an empty line here (we add an empty like between variable declarations and code).

Reply:
also, weird indentation. please stick to multiples of 8 spaces. See CODING_STYLE